boot/kernel_os_test.go

```diff
@@ -214,6 +290,46 @@ func (s *coreBootSetSuite) TestSetNextBootForKernelForTheSameKernelTryMode(c *C)
 	c.Check(reboot, Equals, false)
 }
 
+func (s *coreBootSetSuite) TestSetNextBoot20ForKernelForTheSameKernelTryMode(c *C) {
+	coreDev := boottest.MockUC20Device("pc-kernel")
+	c.Assert(coreDev.HasModeenv(), Equals, true)
+
+	// setup current kernel
+	kernel1, err := snap.ParsePlaceInfoFromSnapFileName("pc-kernel_1.snap")
+	c.Assert(err, IsNil)
+	r := s.bootloader.SetRunKernelImageEnabledKernel(kernel1)
+	defer r()
+
+	bootVars := map[string]string{
+		"kernel_status": "try"}
+	s.bootloader.SetBootVars(bootVars)
+
+	bs := boot.NewCoreBootParticipant(kernel1, snap.TypeKernel, coreDev)
```

Comment:
I think we should also assert that these aren't trivial when we create these, I will do this today